### PR TITLE
Toggle mask boundaries

### LINKED ISSUE #1373: Display (toggle) outlines of masked region
We commonly want to clarify that masking has occurred. One way to do this is to draw outlines around the masked regions.

### PR DESCRIPTION
**Pull request by bnmajor**
This branch allows users to view a boundary of a masked region (polygon/rectangle/ellipse) instead of/in addition to masking the actual data.

Based on [previous discussion](https://github.com/HEXRD/hexrdgui/pull/1588#issuecomment-1766887043) and the open issue, the last remaining tasks for this branch are:

- [x] Add support for setting boundary styling
~- [ ] Do not use intensities in Azimuthal lineout if boundary is visible (even if image values are not being masked)([comment](https://github.com/HEXRD/hexrdgui/issues/1373#issuecomment-1486045904))~

Fixes #1373

### CODE REVIEW

**Comment by bnmajor** [comment]
After further discussion with @psavery and @saransh13 this branch has few further updates:
- The MaskManger UI has been updated to be easier to understand
toggle_mask_bounds
updated to be easier to understand
- Any time that we create/update the polar image we now create/update two: one that represents the image to display, one that represents the data to use computationally
- In places where we retrieve the image for display (stereo plot, polar plot, azimuthal lineout) we grab the display image. Everywhere else we grab the "old" image data (now the computational image).
- Image differences:
    - Computation: Applies all masks to intensity values - "visible" and "boundary"
    - Presentation: Only applies "visible" masks to intensity values. Draws applied boundaries.

File: hexrdgui/image_canvas.py [hunk 1]
```diff
@@ -221,15 +221,15 @@ def unscaled_image_dict(self):
...
+            return {'img': self.iviewer.display_img}
 
     @property
     def unscaled_images(self):
         # Returns a list of the unscaled images
         if self.mode == ViewType.raw:
             return list(self.unscaled_image_dict.values())
         else:
-            return [self.iviewer.img]
+            return [self.iviewer.display_img]
```
[the remark this answers is not in the document]
Comment (psavery) [reply hexrdgui/image_canvas.py:232]: It does look like these are used for computation, so we should probably avoid modifying `unscaled_image_dict` and `unscaled_images` and instead, make a new way that we use to obtain these for display purposes.

File: hexrdgui/image_canvas.py [hunk 2]
```diff
@@ -710,7 +737,7 @@ def show_saturation(self):
             return
 
         # Use the unscaled image data to determine saturation
-        images_dict = self.unscaled_image_dict
+        images_dict = self.unscaled_display_image_dict
```
Comment (bnmajor) [inline hexrdgui/image_canvas.py:740]: @psavery This was one place where I wasn't 100% certain - I am pretty sure we should be using the display image here but I'd like to make sure that's correct

File: hexrdgui/calibration/polarview.py [hunk 3]
```diff
@@ -374,11 +375,16 @@ def apply_image_processing(self):
...
+        # We only apply "visible" masks to the display image
+        img = self.apply_visible_masks(img)
+        disp_img = self.apply_tth_distortion(img)
+        self.display_image = disp_img
 
-        self.processed_img = img
+        # Both "visible" and "boundary" masks are applied to the
+        # computational image
+        comp_img = self.apply_boundary_masks(img)
+        self.computation_img = comp_img
```
[the remark this answers is not in the document]
Comment (bnmajor) [reply hexrdgui/calibration/polarview.py:388]: Fixed :+1:

File: hexrdgui/image_canvas.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bnmajor) [reply hexrdgui/image_canvas.py:249]: > If we are going to use the display_img here, we probably need to add a display_img property to Cartesian and stereo iviewers as well

Fixed!

> It does look like these are used for computation, so we should probably avoid modifying unscaled_image_dict and unscaled_images and instead, make a new way that we use to obtain these for display purposes.

Thanks for the help understanding where I was using the display image when I shouldn't have been! Latest commit adds new `unscaled_display_image_dict`, `unscaled_display_images`, `scaled_display_image_dict`, and `scaled_display_images` properties.

File: hexrdgui/image_canvas.py (hunk 2 above)
Comment (psavery) [reply hexrdgui/image_canvas.py:740]: I think for computing the saturation, we should leave it as the `self.unscaled_image_dict`.

But the saturation is only computed in the raw view anyways, and there's no difference between the display and computed images in the raw view.

Let's just change it back to `self.unscaled_image_dict`

**Pull request merged**